bpf/lib/bpf_cgroup.h

```diff
@@ -281,4 +296,101 @@ get_task_cgroup(struct task_struct *task, __u32 subsys_idx)
 	return cgrp;
 }
 
+/**
+ * get_ancestor_cgroup_id() Returns the ancestor cgroup id of the
+ *    passed cgroup that is at level ancestor_level.
+ * @cgrp: target cgroup
+ * @cgrpfs_ver: Cgroupfs Magic number either Cgroupv1 or Cgroupv2
+ * @ancestor_level: the cgroup ancestor level
+ *
+ * Return id of the cgroup that is the ancestor of the passed cgroup
+ * and is at level ancestor_level, or 0 in case the id could not
+ * be retrieved or the passed cgroup does not have an ancestor at
+ * that level.
+ *
+ * This helper works for both Cgroupv1 and Cgroupv2. The root
+ * cgroup is at ancestor_level zero and each step down the
+ * hierarchy increments the level.
+ *
+ * If ancestor_level == level of passed cgroup, then return value
+ * will be the same as that of get_cgroup_id().
+ */
+static inline __attribute__((always_inline)) __u64
+get_ancestor_cgroup_id(const struct cgroup *cgrp, __u64 cgrpfs_ver,
+		       __u32 ancestor_level)
+{
+	__u32 level;
+	__u64 id = 0;
+
+	if (unlikely(ancestor_level == 0))
+		return id;
+
+#ifdef BPF_FUNC_get_current_ancestor_cgroup_id
+	if (cgrpfs_ver == CGROUP2_SUPER_MAGIC)
+		id = get_current_ancestor_cgroup_id(ancestor_level);
```

Comment:
is there a case where get_current_ancestor_cgroup_id would fail? Could this be,

 return get_current_ancestor_cgroup_id(ancestor_level)

then drop the if (id > 0) below.

Reply:
The bpf_get_current_ancestor_cgroup_id() work on the default hierarchy which is cgroupv2 interface